File: src/main/java/example/springbootdemoserver/person/PersonService.java
package example.springbootdemoserver.person;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Transactional
@Service
public class PersonService {
  @Autowired
  private PersonRepo personRepo;

  public void addPerson(String firstName, String lastName) {
    personRepo.save(new PersonEntity(firstName, lastName));
  }

  public List<PersonDto> list() {
    return personRepo.findAll().stream().map(e -> new PersonDto(e.getFirstName(), e.getLastName())).toList();
  }

  public List<PersonEntity> entities() {
    return personRepo.findAll();
  }
}

File: src/main/kotlin/example/springbootdemoserver/person/PersonService.kt
package example.springbootdemoserver.person

import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Transactional
@Service
class PersonService(private val personRepo: PersonRepo) {
  fun addPerson(firstName: String, lastName: String) =
    personRepo.save(PersonEntity(firstName = firstName, lastName = lastName))

  fun list(): List<PersonDto> =
    personRepo.findAll().map { PersonDto(firstName = it.firstName, lastName = it.lastName) }

  fun entities(): List<PersonEntity> = personRepo.findAll()
}
